packages/admin/admin-generator/src/commands/generate/generateGrid/generateGrid.ts

```diff
@@ -220,26 +223,22 @@ export function generateGrid(
     const iconsToImport: string[] = ["Add", "Edit", "Info", "Excel"];
     const props: Prop[] = [];
 
-    const fieldList = generateGqlFieldList({
-        columns: config.columns.filter((column) => {
-            return (
-                // exclude id because it's always required
-                column.type !== "actions" && column.name !== "id"
-            );
-        }),
+    // Explicitly type the filtered columns to avoid deep type instantiation and "as any[]"
+    const filteredColumns: GridConfigGridColumnDef<Row>[] = config.columns.filter((column) => {
+        return column.type !== "actions" && column.name !== "id";
+    });
+
+    const fieldList = generateGqlFieldList<Row>({
+        columns: filteredColumns,
```

Comment:
I don't understand this change: why is this necessary? Where is the removed "as any[]"?

Reply:
that's not needed anymore due to refactoring. I had here some strange circularity problems, but they are gone now.